src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,196 @@
+                throw new IndexOutOfBoundsException("Index out of bounds");
+            }
+            currentNode = currentNode.next;
+        }
+
+        if (index == 0) {
+            newValue.next = currentNode.next;
+            if (currentNode.next != null) {
+                currentNode.next.prev = newValue;
+            }
+            head = newValue;
+            return currentNode.item;
+        }
+
+        currentNode.prev.next = newValue;
+        newValue.prev = currentNode.prev;
+        newValue.next = currentNode.next;
+        if (currentNode.next != null) {
+            currentNode.next.prev = newValue;
+        }
+
+        return currentNode.item;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = head;
+        for (int i = 0; i < index; i++) {
+            if (currentNode == null || currentNode.next == null) {
+                throw new IndexOutOfBoundsException("Index out of bounds");
+            }
+            currentNode = currentNode.next;
+        }
+        if (index == 0) {
+            if (head.next == null) {
+                T val = head.item;
+                head = null;
+                size--;
+                return val;
+            }
+            head = head.next;
+            head.prev = null;
+            size--;
+            return currentNode.item;
+        }
+        currentNode.prev.next = currentNode.next;
+        if (currentNode.next != null) {
+            currentNode.next.prev = currentNode.prev;
+        }
+        size--;
+        return currentNode.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(object, currentNode.item)) {
+                if (currentNode.prev != null) {
+                    currentNode.prev.next = currentNode.next;
+                } else {
+                    head = currentNode.next;
+                }
+                if (currentNode.next != null) {
+                    currentNode.next.prev = currentNode.prev;
+                } else {
+                    tail = tail.prev;
+                }
+                size--;
+                return true;
+            }
+            currentNode = currentNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndex(int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("index out of bounds");
+        }
+    }
+
+    private static class Node<T> {
+        private final T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(T element) {
+            this.item = element;
+        }
```

Comment:
let's also assign prev and next in constructor. It will help us simplify the code in methods